src/Model/File.php

```diff
@@ -13,34 +13,34 @@ class File extends Model
 
     public ?string $titleField = 'meta_filename';
 
+    /** @const string All uploaded files first get this status */
+    public const STATUS_DRAFT = 'draft';
+    /** @const string When file is linked to some other model */
+    public const STATUS_LINKED = 'linked';
+    /** @const array */
+    public const ALL_STATUSES = [
+        self::STATUS_DRAFT,
+        self::STATUS_LINKED,
+    ];
+
     /** @var Filesystem */
     public $flysystem;
 
-    public function newFile(): Model
-    {
-        $entity = $this->createEntity();
-
-        $entity->set('token', uniqid('token-'));
-        $entity->set('location', uniqid('file-') . '.bin');
-
-        return $entity;
-    }
-
     protected function init(): void
     {
         parent::init();
 
         $this->addField('token', ['system' => true, 'type' => 'string', 'required' => true]);
         $this->addField('location');
-        $this->addField('url');
-        $this->addField('storage');
-        $this->hasOne('source_file_id', [
+        $this->addField('url'); // not implemented
```

Comment:
delete url/storage field

is source_file_id field used now? If not, delete it and let's readd it in the next PR if used there

Reply:
Not used for now, but it was here before and I really do not see the point to remove it for 5 minutes and then put it right back in next PR :)

Reply:
removed unused fields for now